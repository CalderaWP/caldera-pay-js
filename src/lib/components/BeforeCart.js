// @flow
import React from 'react';
import type {Product} from "../types";
import {Button} from '@wordpress/components';

type Props = {
	productsInCart: Array<Product>,
	checkoutLink: string,
	onClose: Function
};

/**
 * Displays a message designed to be used right after adding to cart
 *
 * @todo add a messsage?
 *
 * @param {Props} props
 * @return {*}
 * @constructor
 */
export const BeforeCart = (props: Props) => {
	return (
		<div>
			<a
				href={props.checkoutLink}
			>
				Checkout Now
			</a>
			<Button
				onClick={props.onClose}
			>
				Continue
			</Button>
			<div>Items In Cart: {props.productsInCart.length}</div>
		</div>
	);
};