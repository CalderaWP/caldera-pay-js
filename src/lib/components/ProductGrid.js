// @flow
import React from 'react';
import  classNames from 'classnames';
import type {ColumnHeader, Row} from "../types";

type Props = {
	headers: Array<ColumnHeader>,
	rows: Array<Row>,
};

/**
 * Displays all products in a table
 * @param props
 * @return {*}
 * @constructor
 */
export  const  ProductGrid =  (props: Props) => {
	return(
		<table
			className={classNames([
				'striped', 'bordered', 'condensed', 'hover', 'table'
			])} >
			<thead>
			<tr>
				{props.headers.map((header:ColumnHeader) => {
					if( header.addToCart ){
						return (
							<th
								key={header.key}
								className={header.className}
							>
								<a href={header.addToCart} >
									{header.label}
								</a>
							</th>
						);

					}
					return (
						<th
							key={header.key}
							className={header.className}
						>
							{header.label}
						</th>
					);
				})}
			</tr>
			</thead>
			<tbody>
			{props.rows.map((row: Row)=> {
				return(
					<tr
						key={row.key}
					>
						{Object.values(props.headers).map((headerColumn: ColumnHeader) => {
							return (<th key={headerColumn.key}>{row[headerColumn.key]}</th>)
						})}
					</tr>
				);

			})}

			</tbody>
		</table>
	);
};