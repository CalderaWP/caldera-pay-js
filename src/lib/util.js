import type {CalderaPayProductInfo} from "./types";


/**
 * Check if a product is in a bundle
 * @param {Number}productId
 * @param {CalderaPayProductInfo}bundleProductInfo
 * @return {Boolean}
 */
export const inBundle = (productId: number, bundleProductInfo: CalderaPayProductInfo) : boolean => {
	const bundledDownloads = bundleProductInfo.bundle.includes.map((value: string): number => { return parseInt(value,10)});
	return bundledDownloads.includes(parseInt(productId,10));
};

/**
 * Get the intersection of two numeric arrays
 *
 * @see https://stackoverflow.com/a/37041756/1469799
 *
 * @param {number[]} arrayOne First array
 * @param {number[]} arrayTwo Second array
 * @return {number[]}
 */
export const intersect = (arrayOne: Array<number>, arrayTwo: Array<number>) :  Array<number>  => {
	return [...new Set(arrayTwo)].filter(x => new Set(arrayTwo).has(x));
};

//This is copied from caldera-admin. Choose one location.
export const pickArray = (array, key) => {
	return array.reduce(
		(accumualtor, item) =>
			accumualtor.concat([item[key]]), []
	);
};