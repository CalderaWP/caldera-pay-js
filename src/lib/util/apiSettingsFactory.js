import type {CalderaPaySettings} from "../types";

/**
 * Factory for API settings
 * @param settings
 * @param apiRoot
 * @return {CalderaPaySettings}
 */
export function apiSettingsFactory(settings: CalderaPaySettings,apiRoot:string) {

	function addApiRoot(endpoint:string):string{
		return `${apiRoot}/${endpoint.replace(/^\/|\/$/g, '')}`;
	}

	const nameSpace = 'calderapay/v1';

	function addApiRootAndNamespace(endpoint:string):string{
		return `${apiRoot}/${nameSpace}/${endpoint.replace(/^\/|\/$/g, '')}`;
	}
	settings = {
		...settings,
		productsRoute: addApiRoot('wp/v2/download'),
		cartRoute: addApiRootAndNamespace('cart'),
		embeddedApi: {
			keyRoute: addApiRootAndNamespace( 'pay/qualpay/key'),
			paymentRoute: addApiRootAndNamespace( 'pay/qualpay'),
		},
		checkoutLink: 'https://calderaformscom.lndo.site/checkout',
		bundleOrder: [
			'isFree',
			20520, //Individual
			20518, //Advanced
			48255, //Agency
		]
	};

	return settings;
}