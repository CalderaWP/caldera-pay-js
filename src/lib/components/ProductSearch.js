// @flow
import React from 'react';
import {RenderGroup} from '@caldera-labs/components';


type Props = {
	onProductSearch: Function,
	searchTerm: string
};
export  const ProductSearch = (props : Props) => {

	const searchField = {
		id: 'caldera-pay-product-search',
		label: 'Search Products',
		desc: '',
		type: 'input',
		innerType: 'search',
		inputClass: '',
		value: props.searchTerm,
		onValueChange: (newValue) => {
			props.onProductSearch(newValue);
		}
	};

	return (
		<div>
			<RenderGroup
				configFields={[
					searchField
				]}
				className={'caldera-pay-product-search-options'}
			/>
		</div>
	);

};

