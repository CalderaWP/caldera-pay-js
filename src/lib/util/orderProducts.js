// @flow
import type {Product} from "../types";
import {inBundle, pickArray} from "../util";

export function orderProducts(bundles: Array<Product>, products: Array<Product>, bundleOrder: Array<string|number>) {
	let bundleMap = {};
	bundles.forEach((bundle) => {
		bundleMap[bundle.id] = products.filter(product => inBundle(product.id, bundle.calderaPay));
	});
	bundleMap['isFree'] = products.filter((product: Product) => product.calderaPay.prices.free);
	let ordered: Array<number> = [];

	/**
	 * Concact two arrays of numbers, adding ONLY unique values
	 * @param {(number)[]} currentValues Array to add to. All values of this array will remain.
	 * @param {(number)[]} newValues Values to add to current values. Only values NOT present in currentValues will be added.
	 * @return {(number)[]}
	 */
	const addToArrayUniqueNewOnly = (currentValues: Array<number>, newValues: Array<number>): Array<number> => {
		return Array.from(new Set(currentValues.concat(newValues)));
	};

	bundleOrder.forEach((key: number | string) => {
		ordered = addToArrayUniqueNewOnly(ordered, pickArray(bundleMap[key], 'id'));
	});
	return ordered;
}