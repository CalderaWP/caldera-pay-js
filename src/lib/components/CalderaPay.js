// @flow
import React, {Component} from 'react';
import {ProductGrid} from "./ProductGrid";
import type {
	Product, ColumnHeader, CalderaPaySettings,
	//ProductCollection,
	Row, CalderaPayUserSettings
} from "../types";
import type {WordPressUser} from "../types/WordPress";
import {ProductSearch} from "./ProductSearch";
import Fuse from 'fuse.js';
import {pickArray, intersect, inBundle} from "../util";
import {Spinner} from '@wordpress/components';
import bundles from "../__MOCKDATA__/bundles";
import {User} from "../containers/User";
import {SelectBundle} from "../containers/SelectBundle";
import {ApiClient} from "../api/ApiClient";
import type {ProductsData} from "../api/ApiClient";
import {qualpayEmbeddedFields} from "../qualpayEmbeddedFields";
import type {PaymentItem} from "../types/qualpay";

/**
 * Props type for CalderaPay Component
 */
type Props = {
	settings: CalderaPaySettings,
	userSettings: CalderaPayUserSettings,
	fuseOptions: ?Fuse.FuseOptions,
	apiClient: ApiClient,
	qualpayEmbeddedFields: qualpayEmbeddedFields
};

/**
 * State type for CalderaPay Component
 */
type State = {
	products: Array<Product>,
	bundles: Array<Product>,
	ordered: Array<number>,
	searchTerm: string,
	hasLoaded: boolean,
	user: WordPressUser,
	jwtToken: string,
	productSelectedId: ?number,
	productIdToPurchase: ?number,
	isPaymentOpen: boolean,
	hasPaymentLoaded: boolean,
	cardToken: string,
	purchaseErrors: Array<string>
};


/**
 * The outermost container for CalderaPay UI
 */
export class CalderaPay extends Component<Props, State> {

	/** @inheritDoc **/
	static defaultProps = {
		fuseOptions: {
			shouldSort: false,
			threshold: 0.6,
			location: 0,
			distance: 100,
			maxPatternLength: 32,
			minMatchCharLength: 3,
			keys: [
				'title.rendered',
				'content.rendered',
				'excerpt.rendered'
			]
		},
	};

	loaded: boolean;
	/**
	 * @type State
	 */
	state = {
		products: [],
		bundles: [],
		ordered: [],
		searchTerm: '',
		hasLoaded: false,
		jwtToken: '',
		user: {
			id: 0,
			username: '',
			name: '',
			first_name: '',
			last_name: '',
			email: ''
		},
		productSelectedId: 0,
		productIdToPurchase: 0,
		isPaymentOpen: false,
		hasPaymentLoaded: false,
		cardToken: '',
		purchaseErrors: []
	};

	/** @inheritDoc **/
	constructor(props: Props) {
		super(props);
		(this: any).setSearchTerm = this.setSearchTerm.bind(this);
		(this: any).onValidateToken = this.onValidateToken.bind(this);
		(this: any).setProductSelected = this.setProductSelected.bind(this);
		(this: any).productIdToPurchase = this.productIdToPurchase.bind(this);
		(this: any).onPurchaseError = this.onPurchaseError.bind(this);
		(this: any).onPurchaseSuccess = this.onPurchaseSuccess.bind(this);
		(this: any).getPurchaseTotal = this.getPurchaseTotal.bind(this);

		const {settings,userSettings} = this.props;
	}

	/** @inheritDoc **/
	componentDidMount() {
		const {settings,apiClient} = this.props;
		const {bundleOrder} = settings;
		apiClient.getProducts(bundleOrder).then((data:ProductsData)=> {
			this.setState({
				products:data.products,
				bundles:data.bundles,
				ordered:data.ordered,
				hasLoaded: true
			});
		});


	}


	/**
	 * Get the product table rows of products, in order
	 * @return {Array}
	 */
	getRows(): Array<Row> {
		const rows = [];
		const {products, bundles, ordered} = this.state;
		ordered.forEach((productId: number) => {
			const product: Product = products.find(
				product => productId === parseInt(product.id, 10)
			);
			const {calderaPay} = product;
			const row: Row = {
				key: product.id,
				label: product.title.rendered,
				isFree: calderaPay.prices.free,
				addToCart: calderaPay.prices.addToCart,
				link: product.link,
				product: product
			};

			bundles.forEach((bundle): Product => {
				row[bundle.id] = inBundle(product.id, bundle.calderaPay);
			});
			rows.push(row);
		});

		return rows;

	}

	/**
	 * Get the product table headers, in order.
	 * @return {*[]}
	 */
	getHeaders(): Array<ColumnHeader> {
		const columns = [
			{
				label: 'Add-On Name',
				className: 'sr-only',
				key: 'label',
				id: 0,
				showLabel: true,
			},
			{
				key: 'isFree',
				label: 'Free',
				className: '',
				id: 0,
				showLabel: false,
			},

		];
		const {bundles} = this.state;
		//Bundles not set yet? Bail early
		if (!bundles.length) {
			return columns;
		}

		this.props.settings.bundleOrder
			.filter(x => !isNaN(x))
			.forEach((bundleId: number) => {
				function findBundle(bundleId: number): Product {
					return bundles.find((bundle: Product) => bundle.id === bundleId);
				}

				const bundle = findBundle(bundleId);
				const {calderaPay} = bundle;
				columns.push({
					label: bundle.title.rendered,
					className: '',
					id: bundle.id,
					key: bundle.id,
					addToCart: calderaPay.prices.addToCart,
					bundle: bundle,
					showLabel: false,

				});
			});
		return columns;

	}

	/**
	 * Mutate search term in state
	 *
	 * @param {string} searchTerm
	 */
	setSearchTerm(searchTerm: string) {
		const {ordered} = this.state;
		this.setState({searchTerm});
		//This should be its own method
		const {products} = this.state;
		//No way newing Fuse everytime will be performant.
		//Would need to new it every time this.state.products mutates though
		const result = new Fuse(products, this.props.fuseOptions).search(searchTerm);
		this.setState({ordered: pickArray(intersect(ordered, result), 'id')});

	}

	/**
	 * Set the product that is currently selected
	 *
	 * @param productSelectedId
	 */
	setProductSelected(productSelectedId: number) {
		this.setState({productSelectedId});
	}


	getSelectedProduct(): {
		product: Product,
		bundlesIncludedIn: ?Array<Product>
	} {

		const {products, bundles, productSelectedId} = this.state;
		const product: Product = products.find(
			product => productSelectedId === parseInt(product.id, 10)
		);

		const bundlesIncludedIn: Array<Product> = [];

		this.props.settings.bundleOrder
			.filter(x => !isNaN(x))
			.forEach((bundleId: number) => {
				function findBundle(bundleId: number): Product {
					return bundles.find((bundle: Product) => bundle.id === bundleId);
				}

				const bundle = findBundle(bundleId);
				bundlesIncludedIn.push(bundle);
			});

		return {
			product,
			bundlesIncludedIn,
		}
	}

	/**
	 * Callback when JWT Token is validated
	 *
	 * @param jwtToken
	 */
	onValidateToken(jwtToken: string) {
		this.setState({jwtToken});
	}

	/**
	 * Callback when payment has succeed
	 * @param cardToken
	 */
	onPurchaseSuccess(cardToken: string){
		this.setState({
			isPaymentOpen: false,
			cardToken,
			purchaseErrors: []
		});
	}

	/**
	 * Set payment errors
	 * @param purchaseErrors
	 */
	onPurchaseError(purchaseErrors: Array<string>){
		this.setState({purchaseErrors});
	}

	/**
	 * When product ID to prucahse is set, load paymen
	 * @param productIdToPurchase
	 */
	productIdToPurchase(productIdToPurchase: number) {
		const{hasPaymentLoaded}= this.state;
		const{qualpayEmbeddedFields}= this.props;
		this.setState({
			productIdToPurchase,
			isPaymentOpen: true,
		});
		if(! hasPaymentLoaded ){
			const total = this.getPurchaseTotal();
			qualpayEmbeddedFields
				.putFormOnDom()
				.loadCheckout(
					{
						total,
						displayItems:[
							{
								label: 'Tax',
								amount: {
									currency: 'USD',
									value: 1.00,
								},
							},
							{
								label: 'Subtotal',
								amount: {
									currency: 'USD',
									value: 10.00,
								},
							},
						]
					},
					this.onPurchaseSuccess,
					this.onPurchaseError
				);
		}
	}

	/**
	 * Get the total for the purchase
	 * @return {number}
	 */


	getPurchaseTotal(): PaymentItem
	{
		const {products, productIdToPurchase} = this.state;
		function findPurchaseProduct(): Product {
			return products.find((product: Product) => product.id === productIdToPurchase);
		}

		const product = findPurchaseProduct();
		return {
			label: product.title.rendered,
			amount: {
				currency: 'USD',
				value: product.calderaPay.prices.price
			}
		}
	}


	/** @inheritDoc **/
	render() {
		const {state, props} = this;
		const {searchTerm, hasLoaded, productSelectedId, jwtToken, productIdToPurchase,isPaymentOpen} = state;
		const {userSettings} = props;
		if (!hasLoaded) {
			return <div>
				<div className={'sr-only'}>Loading</div>
				<Spinner/></div>
		}

		if( isPaymentOpen){
			return (
				<div className={'container'}>
					<User
						settings={userSettings}
						jwtToken={jwtToken}
						onValidateToken={this.onValidateToken}
					/>
				</div>
			);
		}
		return (
			<div className={'container'}>
				{productSelectedId &&
				<React.Fragment>
					<div>
						{!productIdToPurchase &&
						<SelectBundle
							product={this.getSelectedProduct().product}
							bundlesIncludedIn={this.getSelectedProduct().bundlesIncludedIn}
							onSelectForPurchase={this.productIdToPurchase}
						/>
						}

					</div>

					{productIdToPurchase &&
					<div className={'row'}>
						<div className={'col-sm-12 col-md-6'}>
							<User
								settings={userSettings}
								jwtToken={jwtToken}
								onValidateToken={this.onValidateToken}
							/>
						</div>

					</div>

					}
				</React.Fragment>

				}

				{!productSelectedId &&
				<React.Fragment>
					<div className={'row'}>
						<div className={'col-md-9'}>
							<ProductSearch
								searchTerm={searchTerm}
								onProductSearch={this.setSearchTerm}
							/>
						</div>
						<div className={'col-md-3'}>
							<User
								settings={userSettings}
								jwtToken={jwtToken}
								onValidateToken={this.onValidateToken}
							/>
						</div>
					</div>

					<div>
						<ProductGrid
							products={state.products}
							rows={this.getRows()}
							headers={this.getHeaders()}
							onAddToCart={this.setProductSelected}
							bundles={bundles}
						/>
					</div>
				</React.Fragment>
				}


			</div>

		);


	}
}


