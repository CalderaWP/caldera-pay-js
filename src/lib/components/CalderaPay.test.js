import  {CalderaPay} from "./CalderaPay";
import React from "react";
import renderer from "react-test-renderer";
import {ProductGrid} from "./ProductGrid";
import {BeforeCart} from "./BeforeCart";
import {orderProducts} from "../util/orderProducts";
import {ProductSearch} from "./ProductSearch";
import type {Row} from "../types";
import {CartOverview} from "./CartOverview";
import bundles from '../__MOCKDATA__/bundles';
import products from '../__MOCKDATA__/products';

const genericHandler = () => {};
const bundleOrder = [
	'isFree',
	20520, //Individual
	20518, //Advanced
	48255, //Agency
];

const ordered = orderProducts(bundles,products,bundleOrder);
describe( 'Main component', () => {

	it.skip( 'Passes snapshot tests', () => {
		const component = renderer.create(<CalderaPay
			settings={{
				apiRoot: 'https://calderaformscom.lndo.site/wp-json/wp/v2/download',
				cartRoute: 'https://calderaformscom.lndo.site/wp-json/calderapay/v1/cart',
				checkoutLink: 'https://calderaformscom.lndo.site/checkout',
				bundleOrder: bundleOrder
			}}

		/>);
		expect( component.toJSON() ).toMatchSnapshot();
	});


});

const cartItems = [
	{
		id: 20520,
	}
];

describe( 'Individual components', () => {
	it( 'BeforeCart snapshot empty cart', () => {
		const component = renderer.create(
			<BeforeCart
				productsInCart={[]}
				checkoutLink={'https://calderaformscom.lndo.site/checkout'}
				onClose={genericHandler}
			/>
		);
		expect(component.toJSON()).toMatchSnapshot();
	});


	it( 'BeforeCart snapshot empty cart', () => {

		const component = renderer.create(
			<BeforeCart
				productsInCart={cartItems}
				checkoutLink={'https://calderaformscom.lndo.site/checkout'}
				onClose={genericHandler}
			/>
		);
		expect(component.toJSON()).toMatchSnapshot();
	});

});

describe( 'CalderaPay components', () => {
	it( 'BeforeCart snapshot empty cart', () => {
		const component = renderer.create(
			<BeforeCart
				productsInCart={[]}
				checkoutLink={'https://calderaformscom.lndo.site/checkout'}
				onClose={genericHandler}
			/>
		);
		expect(component.toJSON()).toMatchSnapshot();
	});


	it( 'BeforeCart snapshot empty cart', () => {
		const cartItems = [
			{
				id:
					20520,
				options:
					{},
			}
		];
		const component = renderer.create(
			<BeforeCart
				productsInCart={cartItems}
				checkoutLink={'https://calderaformscom.lndo.site/checkout'}
				onClose={genericHandler}
			/>
		);
		expect(component.toJSON()).toMatchSnapshot();
	});


	it( 'ProductGrid matches snapshot', () => {
		const headers = [
			{
				label: 'Add-On Name',
				className: 'sr-only',
				key: 'label',
				id: 0
			},
			{
				key: 'isFree',
				label: 'Free',
				className: '',
				id: 0
			},
			{
				label: 'Bundle 1',
				className: '',
				id: 1,
				key: 1,
				addToCart: 'https://cart.com'
			},
			{
				label: 'Bundle 2',
				className: '',
				id: 2,
				key: 2,
				addToCart: 'https://cart.com'
			}

		];

		const rows : Array<Row> = [
			{
				key: 42,
				label: 'Product in bundle 1',
				isFree: 'No',
				addToCart: 'https://cart.com',
				link: 'https://link.com',
				1: true,
				2: false
			},
			{
				key: 43,
				label: 'Product in bundle 1 and 2',
				isFree: 'No',
				addToCart: 'https://cart.com',
				link: 'https://link.com',
				1: true,
				2: true,
			},

		];

			const component = renderer.create(
				<ProductGrid
					headers={headers} rows={rows} onAddToCart={genericHandler} bundles={bundles}/>
			);
			expect(component.toJSON()).toMatchSnapshot();
	});




	it( 'CartOverview matches snapshot with no items in cart', () => {
		const component = renderer.create(
			<CartOverview productsInCart={[]} checkoutLink={'https://link.com'}/>
		);
		expect(component.toJSON()).toMatchSnapshot();


	});

	it( 'CartOverview matches snapshot with  items in cart', () => {
		const component = renderer.create(
			<CartOverview productsInCart={cartItems} checkoutLink={'https://link.com'}/>
		);
		expect(component.toJSON()).toMatchSnapshot();


	});

	it( 'ProductSearch matches snapshot with no search term', () => {
		const component = renderer.create(
			<ProductSearch onProductSearch={genericHandler} searchTerm={''}/>
		);
		expect(component.toJSON()).toMatchSnapshot();

	});


	it( 'ProductSearch matches snapshot with a search term', () => {
		const component = renderer.create(
			<ProductSearch onProductSearch={genericHandler} searchTerm={'MailChimp'}/>
		);
		expect(component.toJSON()).toMatchSnapshot();


	});


});