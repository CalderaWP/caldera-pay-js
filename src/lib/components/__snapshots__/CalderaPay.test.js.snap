// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CalderaPay components BeforeCart snapshot empty cart 1`] = `
<div>
  <a
    href="https://calderaformscom.lndo.site/checkout"
  >
    Checkout  Now
  </a>
  <button
    className="components-button"
    disabled={undefined}
    onClick={[Function]}
    type="button"
  >
    Continue
  </button>
  <div>
    Items In Cart: 
    0
  </div>
</div>
`;

exports[`CalderaPay components BeforeCart snapshot empty cart 2`] = `
<div>
  <a
    href="https://calderaformscom.lndo.site/checkout"
  >
    Checkout  Now
  </a>
  <button
    className="components-button"
    disabled={undefined}
    onClick={[Function]}
    type="button"
  >
    Continue
  </button>
  <div>
    Items In Cart: 
    1
  </div>
</div>
`;

exports[`CalderaPay components Cart matches snapshot with  items in cart 1`] = `
<div>
  <div>
    Items In Cart: 
    1
  </div>
  <div>
    <a
      href="https://link.com"
    >
      Checkout
    </a>
  </div>
</div>
`;

exports[`CalderaPay components Cart matches snapshot with no items in cart 1`] = `
<div>
  <div>
    Items In Cart: 
    0
  </div>
  0
</div>
`;

exports[`CalderaPay components ProductGrid matches snapshot 1`] = `
<table
  className="striped bordered condensed hover table"
>
  <thead>
    <tr>
      <th
        className="sr-only"
      >
        Add-On Name
      </th>
      <th
        className=""
      >
        Free
      </th>
      <th
        className=""
      >
        <a
          href="https://cart.com"
        >
          Bundle 1
        </a>
      </th>
      <th
        className=""
      >
        <a
          href="https://cart.com"
        >
          Bundle 2
        </a>
      </th>
    </tr>
  </thead>
  <tbody>
    <tr>
      <th>
        <button
          className="components-button"
          disabled={undefined}
          onClick={[Function]}
          type="button"
        >
          Product in bundle 1
        </button>
      </th>
      <th>
        <button
          className="components-button"
          disabled={undefined}
          onClick={[Function]}
          type="button"
        >
          No
        </button>
      </th>
      <th>
        <button
          className="components-button"
          disabled={undefined}
          onClick={[Function]}
          type="button"
        />
      </th>
      <th>
        <button
          className="components-button"
          disabled={undefined}
          onClick={[Function]}
          type="button"
        />
      </th>
    </tr>
    <tr>
      <th>
        <button
          className="components-button"
          disabled={undefined}
          onClick={[Function]}
          type="button"
        >
          Product in bundle 1 and 2
        </button>
      </th>
      <th>
        <button
          className="components-button"
          disabled={undefined}
          onClick={[Function]}
          type="button"
        >
          No
        </button>
      </th>
      <th>
        <button
          className="components-button"
          disabled={undefined}
          onClick={[Function]}
          type="button"
        />
      </th>
      <th>
        <button
          className="components-button"
          disabled={undefined}
          onClick={[Function]}
          type="button"
        />
      </th>
    </tr>
  </tbody>
</table>
`;

exports[`CalderaPay components ProductSearch matches snapshot with a search term 1`] = `
<div>
  <div
    className="caldera-pay-product-search-options"
  >
    <div
      className="caldera-config-group"
    >
      <div
        className="caldera-config-field"
      >
        <label
          htmlFor="caldera-pay-product-search"
        >
          Search Products
        </label>
        <input
          aria-describedby={null}
          className="field-config field-config"
          disabled={false}
          id="caldera-pay-product-search"
          onBlur={undefined}
          onChange={[Function]}
          onFocus={undefined}
          required={false}
          type="text"
          value="MailChimp"
        />
        
      </div>
    </div>
  </div>
</div>
`;

exports[`CalderaPay components ProductSearch matches snapshot with no search term 1`] = `
<div>
  <div
    className="caldera-pay-product-search-options"
  >
    <div
      className="caldera-config-group"
    >
      <div
        className="caldera-config-field"
      >
        <label
          htmlFor="caldera-pay-product-search"
        >
          Search Products
        </label>
        <input
          aria-describedby={null}
          className="field-config field-config"
          disabled={false}
          id="caldera-pay-product-search"
          onBlur={undefined}
          onChange={[Function]}
          onFocus={undefined}
          required={false}
          type="text"
          value=""
        />
        
      </div>
    </div>
  </div>
</div>
`;

exports[`Individual components BeforeCart snapshot empty cart 1`] = `
<div>
  <a
    href="https://calderaformscom.lndo.site/checkout"
  >
    Checkout  Now
  </a>
  <button
    className="components-button"
    disabled={undefined}
    onClick={[Function]}
    type="button"
  >
    Continue
  </button>
  <div>
    Items In Cart: 
    0
  </div>
</div>
`;

exports[`Individual components BeforeCart snapshot empty cart 2`] = `
<div>
  <a
    href="https://calderaformscom.lndo.site/checkout"
  >
    Checkout  Now
  </a>
  <button
    className="components-button"
    disabled={undefined}
    onClick={[Function]}
    type="button"
  >
    Continue
  </button>
  <div>
    Items In Cart: 
    1
  </div>
</div>
`;

exports[`Main component Passes snapshot tests 1`] = `
<div>
  <div
    className="sr-only"
  >
    Loading
  </div>
  <span
    className="components-spinner"
  />
</div>
`;
